### PR TITLE
home, service, rest, model, card and prime vue changes

### PR DESCRIPTION
**Pull request by LindemirHoffmann**
I added home, service, rest, card, changes to prime vue and axios

### CODE REVIEW

**Review by iagoreal — APPROVED** [review]

**Review by chenriquedev — APPROVED** [review]

File: Projeto-Stream/src/components/paginator.vue
Comment (eltonls) [inline Projeto-Stream/src/components/paginator.vue:1]: Redundant component, use the paginator from primevue directly

File: Projeto-Stream/src/components/card.vue [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            return `${this.baseImageUrl}${this.imageEndpoint}`;
+        },
+        generateUrl(): string {
+            return `/details/${this.type}/${this.id}`;
+        }
+    }
+}
+</script>
+
+<style scoped>
```
Comment (eltonls) [inline Projeto-Stream/src/components/card.vue]: Delete unused styles

File: Projeto-Stream/src/components/nav-bar.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    route: "/",
+                },
+                {
+                    label: 'Filmes',
+                    route: '/movie',
+                },
+                {
+                    label: 'Séries',
+                    route: '/series',
+                    badge: 3,
```
Comment (eltonls) [inline Projeto-Stream/src/components/nav-bar.vue]: What is badge?

File: Projeto-Stream/src/views/Home/index.vue [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        page: Number(this.$route.query.page) || 1,
+        rows: 20,
+        total_pages: 500,
+        total_results: 10000
+      },
+      itemsList: [] as Array<MovieAndTvShow>
+    }
+  },
+  methods: {
+    fetchItems(params?: {}) {
```
Comment (eltonls) [inline Projeto-Stream/src/views/Home/index.vue]: Generic Naming.

File: Projeto-Stream/src/views/Home/index.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
+<template>
+  <div class="px-3 md:px-10 pt-20 md:pt-24 pb-12 bg-black">
+    <div class="grid grid-cols-2 md:grid-cols-4 gap-4">
+      <div
```
Comment (eltonls) [inline Projeto-Stream/src/views/Home/index.vue:4]: Stop using v-for, use DataView instead.

**Review by eltonls — COMMENTED** [review]
Review the code and make changes.

**Review by eltonls — APPROVED** [review]

File: Projeto-Stream/package.json [hunk 5]
```diff
@@ -14,6 +14,7 @@
     "format": "prettier --write src/"
   },
   "dependencies": {
+    "@primevue/themes": "^4.0.4",
```
Comment (chenriquedev) [inline Projeto-Stream/package.json]: Remove the "^" from dependencies.

File: Projeto-Stream/src/assets/main.css [hunk 6]
```diff
@@ -1,35 +1,9 @@
...
-    display: grid;
-    grid-template-columns: 1fr 1fr;
-    padding: 0 2rem;
-  }
-}
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+[data-p-highlight="true"] {
```
Comment (chenriquedev) [inline Projeto-Stream/src/assets/main.css]: is this working ?

File: Projeto-Stream/src/components/nav-bar.vue [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                </router-link>
+            </template>
+            <template #end>
+                <div class="flex align-items-center gap-2 mr-3">
+                    <router-link to="/favorites">
+                        <a href="/minha-lista"><svg stroke="currentColor" fill="red" stroke-width="0" viewBox="0 0 1024 1024" class="text-pirataflix-clean_red transform transition duration-150 ease-in-out hover:scale-105" height="1.5rem" width="1.5rem" xmlns="http://www.w3.org/2000/svg"><path d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"></path></svg></a>
```
Comment (chenriquedev) [inline Projeto-Stream/src/components/nav-bar.vue]: Change this for <i class="pi pi-heart-fill " style="font-size: 2rem; color:red"></i>

File: Projeto-Stream/src/components/nav-bar.vue [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                    label: 'Séries',
+                    route: '/series',
+                },
+            ]
+        };
+    }
+};
+</script>
+
+<style>
```
Comment (chenriquedev) [inline Projeto-Stream/src/components/nav-bar.vue]: Remove unnecessary css

File: Projeto-Stream/src/views/Home/index.vue [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        page: Number(this.$route.query.page) || 1,
+        rows: 20,
+        total_pages: 500,
+        total_results: 10000
+      },
+      moviesAndTvShows: [] as Array<MovieAndTvShow>
+    }
+  },
+  methods: {
+    getAll(params?: {}) {
```
Comment (chenriquedev) [inline Projeto-Stream/src/views/Home/index.vue]: Generic name

File: Projeto-Stream/src/views/Home/index.vue [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+          if (response?.error) {
+            this.showError()
+          } else {
+            this.moviesAndTvShows = response.results
+          }
+        }
+      })
+      this.service.getAllMoviesAndTvShows(params)
+    },
+    changePage(event: PageState) {
```
Comment (chenriquedev) [inline Projeto-Stream/src/views/Home/index.vue]: missing function return. void.

File: Projeto-Stream/src/views/Home/index.vue [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        page: Number(this.$route.query.page) || 1,
+        rows: 20,
+        total_pages: 500,
+        total_results: 10000
+      },
+      moviesAndTvShows: [] as Array<MovieAndTvShow>
+    }
+  },
+  methods: {
+    getAllMovieAndTv(params?: {}) {
```
Comment (chenriquedev) [inline Projeto-Stream/src/views/Home/index.vue]: missing function return. void

**Review by chenriquedev — APPROVED** [review]
LGTM

**Pull request merged**